src/main/java/org/craftercms/studio/controller/rest/v2/UsersController.java

```diff
@@ -402,13 +402,17 @@ public Result getCurrentUserSsoLogoutUrl() {
 
     @GetMapping(FORGOT_PASSWORD)
     public ResultOne<String> forgotPassword(@NotBlank @RequestParam(value = REQUEST_PARAM_USERNAME) String username) {
+        int delay = studioConfiguration.getProperty(SECURITY_SET_PASSWORD_DELAY, Integer.class);
         try {
+            TimeUnit.SECONDS.sleep(delay);
             ValidationUtils.validateValue(new EsapiValidator(USERNAME), username, REQUEST_PARAM_USERNAME);
             userService.forgotPassword(username);
         } catch (ServiceLayerException e) {
             logger.error("Failed to process forgot password for user '{}'", username, e);
         } catch (ValidationException e) {
             logger.error("Validation error while processing forgot password for user '{}'", username, e);
+        } catch (InterruptedException e) {
```

Comment:
This will cause the request to be ignored silently.
We should either return an error response or have a separate try-catch block to prevent the actual operation from being skipped.